compiler/luci-interpreter/src/kernels/ArgMax.test.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/ArgMax.h"
+#include "kernels/TestUtils.h"
+
+namespace luci_interpreter
+{
+namespace kernels
+{
+namespace
+{
+
+using namespace testing;
+
+TEST(ArgMax, Float)
+{
+  Shape input_shape{1, 1, 1, 4};
+  std::vector<float> input_data{
+      0.1, 0.9, 0.7, 0.3,
+  };
+  Shape dimension_shape{};
+  std::vector<int32_t> dimension_data{
+      3,
+  };
+  Tensor input_tensor = makeInputTensor<DataType::FLOAT32>(input_shape, input_data);
+  Tensor dimension_tensor = makeInputTensor<DataType::S32>(dimension_shape, dimension_data);
+  Tensor output_tensor = makeOutputTensor(DataType::S64);
+
+  Shape output_shape{1, 1, 1};
+
+  ArgMaxParams params{};
+  params.output_type = DataType::S64;
+  ArgMax kernel(&input_tensor, &dimension_tensor, &output_tensor, params);
+  kernel.configure();
+  kernel.execute();
+  EXPECT_THAT(extractTensorData<int64_t>(output_tensor), ::testing::ElementsAreArray({1}));
+  EXPECT_THAT(output_tensor.shape(), output_shape);
```

Comment:
Shouldn't it be like `EXPECT_THAT(output_tensor.shape(), ::testing::ElementsAreArray({1,1,1}))` as the above line?

Reply:
In TFlite argmax kernel test page, it checks both output value and output shape. so, I add both checking. but when I checking output shape, I could only get the value with Shape type. so, I just create the same value with same type and compare it.
@jinevening If I could use a `getTensorShape` function on here, I could use as the above line,`houldn't it be like EXPECT_THAT(getTensorShape(output_tensor), ::testing::ElementsAreArray({1,1,1}))`

Reply:
> when I checking output shape, I could only get the value with Shape type. so, I just create the same value with same type and compare it.

I'm not sure this correctly compares the shapes. For shape checking, we may need a function something like `extractShapeData`, which returns std::vector (similar to `extractTensorData`). Using that function, we can compare the elements one by one with `::testing::ElementsAreArray`, as below.
`EXPECT_THAT(extractShapeData(output_tensor.shape()), ::testing::ElementsAreArray({1,1,1}))`

Or, you can just skip checking the shape and leave it as TODO. In fact, shape checking of other Ops is also not supported yet (the task is in our TODO lists #1088). I think the shape checking of `ArgMax` can be done after shape tests for other Ops are added. :)

Reply:
@jinevening Oh, I got it! I will mark this as TODO as a comment. Thanks!

Reply:
@jinevening 
FYI It is OK to check shapes like @struss did. But it will not tell what exactly differs in shapes. That is why I wanted to implement a custom matcher, which will pretty print the differencies in the shapes.